apps/total-typescript/src/components/exercise-overlay.tsx

```diff
@@ -209,15 +212,14 @@ const DefaultOverlay: React.FC<OverlayProps> = ({handlePlay}) => {
               moduleType: module.moduleType,
               lessonType: exercise._type,
             })
-            completeExercise(exercise.slug.current).then(() => {
-              return handleContinue(
-                router,
-                module,
-                nextExercise,
-                handlePlay,
-                path,
-              )
-            })
+            addProgressMutation.mutate(
+              {lessonSlug: exercise.slug.current},
+              {
+                onSettled: (data, error, variables, context) => {
```

Comment:
mutations have a set of callbacks you can pass in to do stuff on their various events